codalab/lib/server_util.py

```diff
@@ -13,6 +16,37 @@
 from codalab.common import precondition
 
 
+class RateLimitExceededError(Exception):
+    pass
+
+
+def rate_limited(max_calls, per_seconds):
+    def decorate(func):
```

Comment:
Please document this function.

Also, why do we need two parameters instead of a single `max_calls_per_second`?  I find `max_calls` kind of hard to reason about due to the `min` statement below.